lock/lock.go

```diff
@@ -0,0 +1,320 @@
+package lock
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/retry"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	// Terraform requires the DynamoDB table to have a primary key with this name
+	attributeLockId = "LockID"
+
+	// This is used as the value for maximum retries when creating the DynamoDB table
+	// Default is to retry for up to 5 minutes
+	maxRetriesWaitingForTableToBeActive = 30
+	sleepBetweenTableStatusChecks       = 10 * time.Second
+
+	dynamoDbPayPerRequestBillingMode = "PAY_PER_REQUEST"
+)
+
+type Options struct {
+	// The AWS region for which you wish to create the distributed lock for. For example, if set  to `us-east-1`,
+	// the lock will create a DynamoDB table and record for the AWS resource (e.g. Security Hub) in the same region to mark it as locked.
+	// This means if the same AWS resource (Security Hub) is then reference from a different part of the system, also using this lock mechanism,
+	// the resource will appear as it's locked already and therefore not available to modify.
+	AwsRegion string
+	// The name of the DynamoDB table that will store the lock status for the resource in the given region.
+	LockTable string
+	// The name of the DynamoDB Item value that will store the lock status for the resource in the given region.
+	LockString string
+	// The value for how many times should the AcquireLock and ReleaseLock retry to acquire the lock
+	MaxRetries int
+	// The value for how long should the AcquireLock and ReleaseLock sleep for between retries times should the AcquireLock retry to acquire the lock
```

Comment:
Sorry, one more tiny thing: the `ReleaseLock` thing no longer uses retries, so these comments prob need to be updated.

Reply:
Very good catch - I totally missed it! Thank you